kornia/losses/lovasz_hinge.py

```diff
@@ -0,0 +1,149 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# based on:
+# https://github.com/bermanmaxim/LovaszSoftmax
+
+
+def lovasz_hinge_loss(input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    r"""Criterion that computes a surrogate binary intersection-over-union (IoU) loss.
+
+    According to [2], we compute the IoU as follows:
+
+    .. math::
+
+        \text{IoU}(x, class) = \frac{|X \cap Y|}{|X \cup Y|}
+
+    [1] approximates this fomular with a surrogate, which is fully differentable.
+
+    Where:
+       - :math:`X` expects to be the scores of each class.
+       - :math:`Y` expects to be the binary tensor with the class labels.
+
+    the loss, is finally computed as:
+
+    .. math::
+
+        \text{loss}(x, class) = 1 - \text{IoU}(x, class)
+
+    Reference:
+        [1] http://proceedings.mlr.press/v37/yub15.pdf
+        [2] https://arxiv.org/pdf/1705.08790.pdf
+
+    . note::
+        This loss function only supports binary labels. For multi-class labels please
+        use the Lovasz-Softmax loss.
+
+    Args:
+        input: logits tensor with shape :math:`(N, 1, H, W)`.
+        labels: labels tensor with shape :math:`(N, H, W)` with binary values.
+
+    Return:
+        a scalar with the computed loss.
+
+    Example:
+        >>> N = 1  # num_classes
+        >>> input = torch.randn(1, N, 3, 5, requires_grad=True)
+        >>> target = torch.empty(1, 3, 5, dtype=torch.long).random_(N)
+        >>> output = lovasz_hinge_loss(input, target)
+        >>> output.backward()
+    """
+    if not isinstance(input, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(target, torch.Tensor):
+        raise TypeError(f"Target type is not a torch.Tensor. Got {type(target)}")
+
+    if not len(input.shape) == 4:
```

Comment:
use `KORNIA_CHECK_SHAPE(input, ["B", "1", "H", "W"])`

same for the others, and explore the testing utilities
https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L186

Reply:
For checking the shapes `KORNIA_CHECK_SHAPE` is used. However, the other checks are still performed "by hand" because `KORNIA_CHECK` does not work with `torch.jit`...